Generate release notes for v8.3.0

craigtaverner opened the pull request:
Once CI is complete the documentation will be available at:
https://elasticsearch_87294.docs-preview.app.elstc.co/diff

craigtaverner committed:
Generate release notes for v8.3.0

lcawl committed:
[DOCS] Add 8.3 migration file to index

craigtaverner committed:
Fixed version number

lcawl commented on docs/reference/migration/migrate_8_3.asciidoc:
```diff
@@ -0,0 +1,51 @@
+==== Cluster and node setting deprecations
+
+[[configuring_bind_dn_in_an_ldap_or_active_directory_ad_realm_without_corresponding_bind_password_deprecated]]
+.-| Configuring a bind DN in an LDAP or Active Directory (AD) realm without a corresponding bind password is deprecated
```
The `-|` characters are appearing in the output, so they seem to be unnecessary:
```suggestion
.Configuring a bind DN in an LDAP or Active Directory (AD) realm without a corresponding bind password is deprecated
```

craigtaverner replied:
I fixed that by editing the original source 85326.yaml to use the correct `|-` instead of `-|`, in https://github.com/elastic/elasticsearch/pull/87294/commits/2aaa3fa1598cd5a565a8a1e6b1d622e4d90e4266

lcawl commented on docs/reference/migration/migrate_8_3.asciidoc:
```diff
@@ -0,0 +1,51 @@
+[%collapsible]
+====
+*Details* +
+-| For LDAP or AD authentication realms, setting a bind DN (via the `xpack.security.authc.realms.ldap.*.bind_dn` realm setting) without a bind password is a misconfiguration that may prevent successful authentication to the node. In the next major release, nodes will fail to start if a bind DN is specified without a password.
```
Ditto:
```suggestion
For LDAP or AD authentication realms, setting a bind DN (via the `xpack.security.authc.realms.ldap.*.bind_dn` realm setting) without a bind password is a misconfiguration that may prevent successful authentication to the node. In the next major release, nodes will fail to start if a bind DN is specified without a password.
```

lcawl reviewed (commented):
I submitted some suggestions around formatting issues in the migration page

craigtaverner committed:
Fix formatting of deprecation in 85326

craigtaverner committed:
Use asciidoc format for deprecations

craigtaverner commented on docs/reference/migration/migrate_8_3.asciidoc:
```diff
@@ -0,0 +1,51 @@
+-| For LDAP or AD authentication realms, setting a bind DN (via the `xpack.security.authc.realms.ldap.*.bind_dn` realm setting) without a bind password is a misconfiguration that may prevent successful authentication to the node. In the next major release, nodes will fail to start if a bind DN is specified without a password.
+
+*Impact* +
+-| If you have a bind DN configured for an LDAP or AD authentication realm, set a bind password for [LDAP](https://www.elastic.co/guide/en/elasticsearch/reference/current/ldap-realm.html#ldap-realm-configuration) or [Active Directory](https://www.elastic.co/guide/en/elasticsearch/reference/current/active-directory-realm.html#ad-realm-configuration). Configuring a bind DN without a password genera ...
```
I fixed that in the original source 85326.yaml in https://github.com/elastic/elasticsearch/pull/87294/commits/960ce34b9dc05856df1855c0edae43152010bb7d

craigtaverner committed:
Extract static content from migration/index

craigtaverner committed:
This was just an enhancement

craigtaverner committed:
Nope, it was an upgrade

craigtaverner committed:
Added migration/index.asciidoc generation support (#87318)

Including extracting static content from migration/index, so the template would be as light as possible.

The reason for this work is because the gradle task `generateReleaseNotes` was not correctly adding new links and imports to the migrations/index and that caused documentation to fail building for 8.3.0.

lcawl committed:
[DOCS] Add ml-cpp PRs to release notes

lcawl reviewed (approved):
I added the ml-cpp PRs from https://github.com/elastic/ml-cpp/blob/8.3/docs/CHANGELOG.asciidoc and the overall layout LGTM, thanks!

craigtaverner committed:
Merge remote-tracking branch 'origin/8.3' into 8.3-release-notes

craigtaverner committed:
Added back incorrectly deleted changlog

craigtaverner committed:
Added missing highlight

craigtaverner committed:
Fixed spelling of StackOverflowError
